### PR TITLE
Add Maven goal preparations and version helpers

### PR DESCRIPTION
**Pull request by cdupuis**
Moved this from atomist-sdm to here.

### CODE REVIEW

**Comment by claymccoy** [comment]
I had a local branch similar to this waiting to become a PR once the maven support was tested in prod /atomist/atomist-sdm#48. I'd tested quite a bit locally, but some things like uploading docker images didn't fully work locally.
So I approve of this, but I'd like to point out some changes that I made to the MavenBuilder functionality:
* It defaults to running the unit tests unless specifically told to skip them. This is a change to the default behavior.
* It will try to use the Maven wrapper if it exists.
* The error finder that looks for "[ERROR]" in the log was removed. I don't know what depends on that, but in my experience, it fails successful builds.

File: lib/maven/build/MavenBuilder.ts [hunk 1]
```diff
@@ -102,3 +100,22 @@ class UpdatingBuild implements LocalBuildInProgress {
     }
 
 }
+
+export async function mavenPackage(p: GitProject,
+                                   progressLog: ProgressLog,
+                                   args: Array<{ name: string, value?: string }> = []): Promise<ChildProcessResult> {
+    const useMavenWrapper = hasMavenWrapper(p);
+    const mavenExec = useMavenWrapper ? "./mvnw" : "mvn";
```
Comment (lievendoclo) [inline lib/maven/build/MavenBuilder.ts]: This won't work on Windows

File: lib/maven/build/MavenBuilder.ts [hunk 2]
```diff
@@ -52,7 +56,9 @@ export class MavenBuilder extends LocalBuilder implements LogInterpretation {
     public logInterpreter: InterpretLog = MavenLogInterpreter;
 
     constructor(sdm: SoftwareDeliveryMachine,
-                private readonly skipTests: boolean = true) {
+                private readonly skipTests: boolean = false,
+                private readonly deploymentUnitFileLocator: (p: LocalProject, mpi: VersionedArtifact) => string =
```
[the remark this answers is not in the document]
Comment (cdupuis) [reply lib/maven/build/MavenBuilder.ts]: Not sure why that is, cc @claymccoy?

[the remark this answers is not in the document]
Comment (claymccoy) [reply lib/maven/build/MavenBuilder.ts]: The original code used a MavenProjectIdentifier. After I made it a separate function it was nicer to use a more general VersionedArtifact.

**Pull request merged**